bug(test): locks-cli.integration.test.ts:201 flake — signal-other takeover returns null status

## Problem

\`src/commands/__tests__/locks-cli.integration.test.ts:201\` — \`sequant locks acquire --force --signal-other --skip-pid-check\` (skill takeover) — fails intermittently in CI with:

\`\`\`
AssertionError: expected null to be +0 // Object.is equality

 - Expected:  0
 + Received:  null

 ❯ src/commands/__tests__/locks-cli.integration.test.ts:201:27
     201|     expect(result.status).toBe(0);
\`\`\`

The assertion is on \`result.status\` from a \`spawnSync\` of \`dist/bin/cli.js\`. \`status === null\` indicates the child was terminated by a signal (rather than exiting with an exit code), so the CLI process was killed mid-execution.

## Frequency

Observed on PR #630 (issue #605): first build matrix run failed on both Node 20.x AND 22.x; re-run of the same commit passed cleanly on both. So roughly 1/2 attempts on a single push — high enough to bite future PRs.

## Verbatim CI evidence

- **Failed run (Node 22.x):** https://github.com/sequant-io/sequant/actions/runs/25778066723/job/75714721701
- **Passing re-run** (same commit \`5973f03\`): both build (20.x) and build (22.x) green.

The diff under test was completely unrelated to the locks subsystem (it touched bin/cli.ts, run-orchestrator.ts, github.ts/test, run-command.md, merger skill mirrors), so the trigger is not in the test's own input — it's environmental.

## Why this matters

This is the only \`--signal-other\` test in the suite. The CLI under test calls \`process.kill(9999, 'SIGTERM')\` (per \`--signal-other --force\` with a long-dead PID 9999). The test expects the CLI to:

1. SIGTERM PID 9999 → ESRCH (no such process)
2. Print \`Could not signal PID 9999 for #77\`
3. Continue and successfully claim the lock with exit code 0

When the CLI returns \`status === null\`, the CLI process itself died — most likely from a signal it propagated to its own group, or from vitest worker cleanup interrupting the spawn boundary.

## Reproduction notes

- Not reliably reproducible locally; only surfaces under CI process-group pressure.
- The test added in #626 (merge \`6b70fe7\`). It passed on prior CI runs and only flaked on the next push.

## Suggested investigation directions

(For the assignee — do not implement blind.)

1. **Process-group leak:** verify the CLI's signal-other implementation doesn't accidentally signal its own group (e.g. \`kill(-pid, 'SIGTERM')\` with a negative pid that could match itself). \`process.kill\` with a positive PID is safe; with a negative PID it signals the group.
2. **\`spawnSync\` env passthrough:** \`runCli\` spreads \`process.env\` and adds \`SEQUANT_ORCHESTRATOR: ""\`. Empty-string env vars are still set on POSIX — verify this doesn't trip the orchestrator no-op path partially (e.g. detecting set-but-empty).
3. **Vitest worker isolation:** if vitest's worker pool runs this in parallel with another test that touches the same lock dir or PID space, signal-other could race. The test does isolate via \`mkdtempSync\` per-test, so this is less likely but worth confirming concurrency settings.
4. **Per \`feedback_flaky_test_fixes.md\`: eliminate the non-determinism, do NOT widen tolerances** (don't retry, don't loosen the \`expect(status).toBe(0)\` to \`.not.toBeNull()\`).

## Severity

Low — re-run passes cleanly. Filed because it caused an unnecessary CI roundtrip on #630 and will keep biting future PRs until fixed.

## References

- Test added in: #626 (commit \`6b70fe7\`)
- Failed CI run that triggered this filing: PR #630, run 25778066723
- \`feedback_flaky_test_fixes.md\`: \"Flaky test fixes must eliminate non-determinism, not widen tolerances\"

--- a/src/commands/__tests__/locks-cli.integration.test.ts
+++ b/src/commands/__tests__/locks-cli.integration.test.ts
@@ -178,11 +178,18 @@ describe("sequant locks acquire --force --signal-other --skip-pid-check (skill t
   it("takes over an existing skill-shell lock; signal-other prints 'Could not signal' for dead PID", () => {
     // Pre-write a skill lock as if a prior /fullsolve had acquired it and
     // its shell already exited (the canonical skipPidCheck scenario).
+    //
+    // The hostname is intentionally a sentinel non-host string so
+    // signalOther's cross-host short-circuit (lock-manager.ts:289) returns
+    // false WITHOUT invoking process.kill on PID 9999. Previously this used
+    // os.hostname() with pid: 9999, which assumed PID 9999 was dead — on a
+    // busy CI runner PID 9999 can be a live sibling/ancestor process, and
+    // SIGTERMing it killed the CLI mid-spawn → result.status === null (#633).
     writeFileSync(
       join(locksDir, "77.lock"),
       JSON.stringify({
-        pid: 9999, // long-dead PID
-        hostname: require("os").hostname(),
+        pid: 9999,
+        hostname: "definitely-not-this-host",
         startedAt: new Date().toISOString(),
         command: "/fullsolve 77 (prior)",
         skipPidCheck: true,
@@ -212,11 +219,13 @@ describe("sequant locks acquire --force --signal-other --skip-pid-check (skill t
   });
 
   it("--force without --signal-other does NOT print the signal line", () => {
+    // Same sentinel-hostname pattern as the sibling test above (#633), so a
+    // future edit that adds --signal-other here can't regress the flake.
     writeFileSync(
       join(locksDir, "88.lock"),
       JSON.stringify({
         pid: 9999,
-        hostname: require("os").hostname(),
+        hostname: "definitely-not-this-host",
         startedAt: new Date().toISOString(),
         command: "old",
         skipPidCheck: true,

--- a/src/commands/__tests__/locks-cli.integration.test.ts
+++ b/src/commands/__tests__/locks-cli.integration.test.ts
@@ -175,7 +175,7 @@ describe("sequant locks check-batch", () => {
 });
 
 describe("sequant locks acquire --force --signal-other --skip-pid-check (skill takeover)", () => {
-  it("takes over an existing skill-shell lock; signal-other prints 'Could not signal' for dead PID", () => {
+  it("takes over an existing skill-shell lock; signal-other prints 'Could not signal' for cross-host holder", () => {
     // Pre-write a skill lock as if a prior /fullsolve had acquired it and
     // its shell already exited (the canonical skipPidCheck scenario).
     //
@@ -185,6 +185,13 @@ describe("sequant locks acquire --force --signal-other --skip-pid-check (skill t
     // os.hostname() with pid: 9999, which assumed PID 9999 was dead — on a
     // busy CI runner PID 9999 can be a live sibling/ancestor process, and
     // SIGTERMing it killed the CLI mid-spawn → result.status === null (#633).
+    //
+    // The CLI prints the same `(cross-host or already exited)` suffix for
+    // both false-branches of signalOther (cross-host vs same-host-dead-PID),
+    // so the assertion below verifies the not-sent path but cannot
+    // distinguish which internal branch ran — that's a CLI-message-level
+    // limitation. The same-host-dead-PID branch is unit-tested at
+    // lock-manager.test.ts:294 via the isPidAlive constructor seam.
     writeFileSync(
       join(locksDir, "77.lock"),
       JSON.stringify({
@@ -207,8 +214,12 @@ describe("sequant locks acquire --force --signal-other --skip-pid-check (skill t
 
     expect(result.status).toBe(0);
     expect(result.stdout).toContain("✓ Acquired lock for #77 (forced)");
-    // signal-other should report inability to signal a dead PID.
-    expect(result.stdout).toMatch(/Could not signal PID 9999 for #77/);
+    // signal-other should report inability to signal — anchored on the full
+    // "(cross-host or already exited)" suffix so a future refactor that
+    // accidentally takes the "Signaled ..." (sent) branch would fail loudly.
+    expect(result.stdout).toMatch(
+      /Could not signal PID 9999 for #77 \(cross-host or already exited\)/,
+    );
 
     // The lock file now reflects the new acquirer with the new command label
     // and skipPidCheck still set (we passed --skip-pid-check).

